db/version_set.cc

```diff
@@ -5191,19 +5208,7 @@ Status VersionSet::ReduceNumberOfLevels(const std::string& dbname,
   }
 
   if (first_nonempty_level > 0) {
-    auto& new_last_level = new_files_list[new_levels - 1];
-
-    new_last_level = vstorage->LevelFiles(first_nonempty_level);
-
-    for (size_t i = 0; i < new_last_level.size(); ++i) {
-      const FileMetaData* const meta = new_last_level[i];
-      assert(meta);
-
-      const uint64_t file_number = meta->fd.GetNumber();
-
-      vstorage->file_locations_[file_number] =
-          VersionStorageInfo::FileLocation(new_levels - 1, i);
-    }
```

Comment:
We do not update the index after `ReduceNumberOfLevels`?

Reply:
It will be done by `PrepareForVersionAppend`, which is done as part of the `LogAndApply` call below.

Reply:
Oh I see. Thanks for the clarification.

Reply:
Maybe some comment about the index not consistent at this point will be informative.

Reply:
Looked into this a bit more and realized that this method is only called by `ldb reduce_levels`, and the DB is not even open during the call (which I suppose is why manipulating the current version directly is "safe"). Still, I tend to think it would be nice to clear the index here and let `LogAndApply` and friends rebuild it from scratch.

Reply:
Actually, on second (third? fourth?) thought, `LogAndApply` creates a new `Version` by "applying" an empty `VersionEdit`, so the contents of the index in the original `Version` do not matter at all. So we might as well leave this as-is for consistency